cli/cli.go

```diff
@@ -1,456 +1,822 @@
+		if err := runPromptActions(p, o,
+			inputCustomNetworkConfig,
+			inputCustomGenesis,
+			inputCustomDeployBlock,
+			inputConsensusBootNodes,
+		); err != nil {
+			return err
+		}
+	} else {
+		if err := inputCheckpointSyncURL(p, o); err != nil {
+			return err
+		}
+		if configs.SupportsMEVBoost(o.genData.Network) {
+			if err := inputMevBoostEndpoint(p, o); err != nil {
+				return err
+			}
+		}
 	}
-	if flags.consensusName != "" {
-		consensusParts := strings.Split(flags.consensusName, ":")
-		// TODO: avoid edit flag
-		flags.consensusName = consensusParts[0]
-		clientImages.consensus = strings.Join(consensusParts[1:], ":")
+	if err := runPromptActions(p, o,
+		inputFeeRecipient,
+		confirmExposeAllPorts,
+		inputGenerationPath,
+	); err != nil {
+		return err
 	}
-	if flags.validatorName != "" {
-		validatorParts := strings.Split(flags.validatorName, ":")
-		// TODO: avoid edit flag
-		flags.validatorName = validatorParts[0]
-		clientImages.validator = strings.Join(validatorParts[1:], ":")
+	if err := setupJWT(p, o, true); err != nil {
+		return err
 	}
+	if err := a.Generate(actions.GenerateOptions{
+		GenerationData: &o.genData,
+		GenerationPath: o.generationPath,
+	}); err != nil {
+		return err
+	}
+	return postGenerate(p, o, a)
+}
 
-	if err := configs.ValidateLoggingFlag(flags.logging); err != nil {
-		return nil, err
+func setupValidatorNode(p prompter.Prompter, o *CliCmdOptions, a actions.SedgeActions) error {
+	if err := selectValidatorClient(p, o); err != nil {
+		return err
+	}
+	if o.genData.Network == NetworkCustom {
+		if err := runPromptActions(p, o,
+			inputCustomNetworkConfig,
+			inputCustomGenesis,
+			inputCustomDeployBlock,
+			inputConsensusBootNodes,
+		); err != nil {
+			return err
+		}
 	}
+	if err := runPromptActions(p, o,
+		inputGraffiti,
+		inputValidatorGracePeriod,
+		inputFeeRecipient,
+		inputGenerationPath,
+	); err != nil {
+		return err
+	}
+	if err := a.Generate(actions.GenerateOptions{
+		GenerationData: &o.genData,
+		GenerationPath: o.generationPath,
+	}); err != nil {
+		return err
+	}
+	return postGenerate(p, o, a)
+}
 
-	// validate custom network flags
-	urlOrPaths := map[string]string{
-		"ChainSpec":      flags.customChainSpec,
-		"Network config": flags.customNetworkConfig,
-		"Genesis":        flags.customGenesis,
+func setupJWT(p prompter.Prompter, o *CliCmdOptions, skip bool) error {
+	if skip {
+		if err := selectJWTSourceOrSkip(p, o); err != nil {
+			return err
+		}
+	} else {
+		if err := selectJWTSource(p, o); err != nil {
+			return err
+		}
 	}
-	for kind, value := range urlOrPaths {
-		if value == "" {
-			continue
+	switch o.jwtSourceType {
+	case SourceTypeCreate:
+		jwtPath, err := generateJWTSecret("")
+		o.genData.JWTSecretPath = jwtPath
+		if err != nil {
+			return err
 		}
-		if err := utils.CheckUrlOrPath(value); err != nil {
-			return nil, fmt.Errorf("invalid %s: %w", kind, err)
+	case SourceTypeExisting:
+		if err := inputJWTPath(p, o); err != nil {
```

Comment:
Call `generateJWTSecret` with the path given by the user. When you pass a path to `generateJWTSecret` it will validate if in the given path exists a jwtsecret file (check introduced in [224](https://github.com/NethermindEth/sedge/pull/224)). If the path is wrong, put the user in a loop until it gives a correct path or it chooses to create the jwtsecret.